### ISSUE TITLE
Looping over undefined when using transport

### ISSUE THREAD

**Issue by jly36963** [pinojs/pino#1805]
## Description

When running the following code

```js
import Koa from "koa";
import pino from "koa-pino-logger";
import json from "koa-json";
import bodyParser from "koa-bodyparser";

// Breaks if caller not specified
const app = new Koa()
  .use(json())
  .use(bodyParser())
  .use(
    pino({
      transport: {
        target: "pino-pretty",
        // caller: "abc"
      },
    })
  );

app.listen();
```

I get the following error.

```
137 |     let fixTarget
138 | 
139 |     for (const filePath of callers) {
                               ^
TypeError: undefined is not an object (evaluating 'callers')
      at fixTarget (<removed>/node_modules/pino-http/node_modules/pino/lib/transport.js:139:27)
```

When I look at the [code block in the error](https://github.com/pinojs/pino/blob/master/lib/transport.js#L129), I see:
- `fixTarget` is trying to loop over `callers`. 
- In this case, I did not provide caller, and [getCaller](https://github.com/pinojs/pino/blob/master/lib/caller.js#L14) is returning `undefined`.
- `callers` is `string | string[] | undefined` but [fixTarget](https://github.com/pinojs/pino/blob/master/lib/transport.js#L77) doesn't handle the `undefined` part of the type union. 
- the `for` loop above breaks when trying to iterate over `undefined`.

## Context

- MacOS 13.5.1 Ventura
- bun 1.0.0
- pino ^8.14.2 (same error when using ^8.15.1)
- pino-pretty  ^10.2.0
- koa ^2.14.2
- koa-pino-logger ^4.0.0

## Workaround

If I un-comment the bogus line `// caller: "abc"`, I circumvent the error

**Comment by jly36963** [comment]
Can't guarantee I will keep this example up forever, but I experienced the issue in [this mini project](https://github.com/jly36963/example-projects/blob/master/ts/bun-koa-kyseley-example/index.ts#L12)

**Comment by jsumners** [comment]
Thank you for the report. This project targets the [Node.js](https://nodejs.org) runtime. If you are encountering issues using Bun, please file issues there as [requested by them](https://twitter.com/jarredsumner/status/1700497869160268155).

**Comment by jly36963** [comment]
@jsumners While using Bun is what surfaced the error, IMO the issue is related to pino. The inner function (`getCaller`) returns `string | string[] | undefined` while the outer function (`fixTarget`) only handles the value as `string | string[]`.

Should I read through the code and make a PR with a fix, or will it be considered off-topic?

I'm traveling in another country currently, but when I'm home I can work on a fix. (I promise I'm not trying to be one of the unhelpful/complaining issue authors 😂 )

**Comment by jsumners** [comment]
`getCallers` is returning `undefined` because Bun's underlying VM does not support `(new Error).stack`. This is discussed in #1804. Thus, when the library is used within a supported runtime, `undefined` being returned from `getCallers` is not possible.

We welcome any PR that will improve the code. If you determine such an improvement, feel free to submit a PR.

**Comment by jly36963** [comment]
Thanks for the feedback.

Even if it's something like
```js
const callers = getCallers()
if (callers === undefined) {
  throw new Error('Only node runtime is supported')
}
```
or `getCallers` throwing an error itself in the `undefined` scenario would be helpful.

It would:
- keep types aligned (no more unhandled `undefined` code path)
- keep annoying people like me from asking these questions.

I'm interested in:
- the success of Bun
- strict typing (and failing fast on unexpected types/values)
- straightforward communication (explicit error instead of confusing/ambiguous error downstream)

If I can be helpful in contributing to Pino, especially for any of the above points, let me know. (I used to contribute to yargs, and maybe I also need to look there for similar issues that pop up when using Bun.)

**Comment by jsumners** [comment]
@mcollina may be able to provide some insight here.

**Comment by christophemarois** [comment]
As an update for interested future readers, this will apparently be fixed in bun `1.0.3` https://github.com/oven-sh/bun/issues/4280#issuecomment-1727666887

**Comment by jly36963** [comment]
@christophemarois heck yes, thank you
